### DISCUSSION TITLE
namespaces for 'RendererOptions' and 'TextGraphicsOptions' could not be found (are you missing a using directive or an assembly reference?)

### DISCUSSION THREAD

**Discussion by bytesfrombits** [SixLabors/ImageSharp#2138] [Ideas]
In order to generate webp images and after upgrading a working project that was generating png images from SixLabors.ImageSharp 1.0.3 and SixLabors.ImageSharp.Drawing 1.0.0-beta11 to SixLabors.ImageSharp 2.1.0 and SixLabors.ImageSharp.Drawing 1.0.0-beta14 respectively, the following errors are now thrown on any attempt to build the project:

Errors:
error CS0246: The type or namespace name 'RendererOptions' could not be found (are you missing a using directive or an assembly reference?)
error CS0246: The type or namespace name 'TextGraphicsOptions' could not be found (are you missing a using directive or an assembly reference?)

Snippet of the affected function:
private static IImageProcessingContext DrawScaleCenterText(this IImageProcessingContext processingContext,Font font,string text,Color color,float padding, int VerticalPosition)
{
Size imgSize = processingContext.GetCurrentSize();

        float targetWidth = imgSize.Width - (padding * 2);
        float targetHeight = imgSize.Height - (padding * 2);

        // measure the text size
        FontRectangle size = TextMeasurer.Measure(text, new RendererOptions(font));

        //find out how much we need to scale the text to fill the space (up or down)
        float scalingFactor = Math.Min(imgSize.Width / size.Width, imgSize.Height / size.Height);

        //create a new font
        Font scaledFont = new Font(font, scalingFactor * font.Size);

        //var center = new PointF(imgSize.Width / 2, imgSize.Height / 2);
        var center = new PointF(imgSize.Width / 2, VerticalPosition);
        var textGraphicOptions = new TextGraphicsOptions()
        {
            TextOptions = {
                HorizontalAlignment = HorizontalAlignment.Center,
                VerticalAlignment = VerticalAlignment.Center
            }
        };
        return processingContext.DrawText(textGraphicOptions, text, font, color, center);
    }

Where/How to reference the RendererOptions, TextGraphicsOptions, FontRectangle, TextMeasurer, TextOptions classes?

The samples in Samples/ImageSharp/ don't appear to be current. Is it possible to update these with the current features such as Wepb support. This would be of great help to other users who may be in the process of updating their projects.

**Comment by JimBobSquarePants** [comment]
Did you read the API docs?

Obviously not since you decided to waste our time by using the Ideas discussion forum for basic QA.

**Reply by JimBobSquarePants** [reply]
Are you kidding me with this? 

https://github.com/SixLabors/ImageSharp/issues/2093
